### ISSUE TITLE
crystal: fix formula so that it doesn’t cause brew installs on Linux arm to fail

### ISSUE THREAD

**Issue by xrisk** [Homebrew/homebrew-core#105566]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
brew config:


HOMEBREW_VERSION: 3.5.4-2-g1294ae6-dirty
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1294ae670473872c53d4b73c5ca843ee97bb5b85
Last commit: 14 minutes ago
Core tap: N/A
HOMEBREW_PREFIX: /home/xrisk/homebrew
HOMEBREW_REPOSITORY: /home/xrisk/homebrew
HOMEBREW_CELLAR: /home/xrisk/homebrew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /home/xrisk/.rbenv/versions/2.6.8/bin/ruby
CPU: quad-core 64-bit arm
Clang: N/A
Git: 2.34.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 5.15.0-40-generic aarch64 GNU/Linux
OS: Ubuntu 22.04 LTS (jammy)
Host glibc: 2.35
/usr/bin/gcc: 11.2.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@5: N/A
xorg: N/A
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.

brew update fails

- [ ] I have resolved all warnings from `brew doctor` and that did not fix my problem. 

brew doctor does not work

### What were you trying to do (and why)?

Install Homebrew on Arm64 Linux.

### What happened (include all command output)?

Source discussion: https://github.com/Homebrew/discussions/discussions/3422

TLDR: https://github.com/Homebrew/homebrew-core/blob/49f24f23b3c43d5fc5a324173dfa2bf1d3cf6f38/Formula/crystal.rb#L74 seems to cause the simple act of _Homebrew installation_ on arm64 Linux to fail. 

```
xrisk@rishav-ubuntu:~$ HOMEBREW_NO_AUTO_UPDATE=1  brew tap xrisk/tap --debug
/home/xrisk/homebrew/Library/Homebrew/shims/shared/git --version
==> Tapping xrisk/tap
Cloning into '/home/xrisk/homebrew/Library/Taps/xrisk/homebrew-tap'...
remote: Enumerating objects: 8, done.
remote: Counting objects: 100% (8/8), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 8 (delta 1), reused 8 (delta 1), pack-reused 0
Receiving objects: 100% (8/8), done.
Resolving deltas: 100% (1/1), done.
/home/xrisk/homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /home/xrisk/homebrew/Library/Taps/xrisk/homebrew-tap/Formula/crystal.rb
Error: Invalid formula: /home/xrisk/homebrew/Library/Taps/xrisk/homebrew-tap/Formula/crystal.rb
kek
/home/xrisk/homebrew/Library/Homebrew/checksum.rb:13:in `initialize'
/home/xrisk/homebrew/Library/Homebrew/resource.rb:201:in `new'
/home/xrisk/homebrew/Library/Homebrew/resource.rb:201:in `sha256'
/home/xrisk/homebrew/Library/Taps/xrisk/homebrew-tap/Formula/crystal.rb:75:in `block in <class:Crystal>'
/home/xrisk/homebrew/Library/Homebrew/resource.rb:42:in `instance_eval'
/home/xrisk/homebrew/Library/Homebrew/resource.rb:42:in `initialize'
/home/xrisk/homebrew/Library/Homebrew/software_spec.rb:106:in `new'
/home/xrisk/homebrew/Library/Homebrew/software_spec.rb:106:in `resource'
/home/xrisk/homebrew/Library/Homebrew/formula.rb:2745:in `block in resource'
/home/xrisk/homebrew/Library/Homebrew/formula.rb:2744:in `each'
/home/xrisk/homebrew/Library/Homebrew/formula.rb:2744:in `resource'
/home/xrisk/homebrew/Library/Taps/xrisk/homebrew-tap/Formula/crystal.rb:60:in `<class:Crystal>'
/home/xrisk/homebrew/Library/Taps/xrisk/homebrew-tap/Formula/crystal.rb:1:in `block in load_formula'
/home/xrisk/homebrew/Library/Homebrew/formulary.rb:90:in `module_eval'
/home/xrisk/homebrew/Library/Homebrew/formulary.rb:90:in `block in load_formula'
/home/xrisk/homebrew/Library/Homebrew/formulary.rb:102:in `load_formula'
/home/xrisk/homebrew/Library/Homebrew/formulary.rb:122:in `load_formula_from_path'
/home/xrisk/homebrew/Library/Homebrew/formulary.rb:309:in `load_file'
/home/xrisk/homebrew/Library/Homebrew/formulary.rb:299:in `klass'
/home/xrisk/homebrew/Library/Homebrew/formulary.rb:294:in `get_formula'
/home/xrisk/homebrew/Library/Homebrew/formulary.rb:543:in `factory'
/home/xrisk/homebrew/Library/Homebrew/readall.rb:45:in `block in valid_formulae?'
/home/xrisk/homebrew/Library/Homebrew/readall.rb:44:in `each'
/home/xrisk/homebrew/Library/Homebrew/readall.rb:44:in `valid_formulae?'
/home/xrisk/homebrew/Library/Homebrew/readall.rb:77:in `valid_tap?'
/home/xrisk/homebrew/Library/Homebrew/tap.rb:311:in `install'
/home/xrisk/homebrew/Library/Homebrew/cmd/tap.rb:65:in `tap'
/home/xrisk/homebrew/Library/Homebrew/brew.rb:93:in `<main>'
Error: Cannot tap xrisk/tap: invalid syntax in tap!
/home/xrisk/homebrew/Library/Homebrew/tap.rb:312:in `install'
/home/xrisk/homebrew/Library/Homebrew/cmd/tap.rb:65:in `tap'
/home/xrisk/homebrew/Library/Homebrew/brew.rb:93:in `<main>’
```

### What did you expect to happen?

Installation to go through.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
git clone https://github.com/Homebrew/brew homebrew 
eval "$(homebrew/bin/brew shellenv)"
brew update —-force
```

I realize that Linux/arm64 is not a well supported configuration. I’m willing to make any required patches myself, please let me know what the appropriate fixes to make in that case are.
